input_writer.go

```diff
@@ -0,0 +1,30 @@
+package logrus
+
+import (
+	"bufio"
+	"io"
+	"runtime"
+)
+
+func (logger *Logger) InputWriter() (*io.PipeWriter) {
+	inputReader, inputWriter := io.Pipe()
+
+	go logger.inputWriterScanner(inputReader)
+	runtime.SetFinalizer(inputWriter, inputWriterFinalizer)
+
+	return inputWriter
+}
+
+func (logger *Logger) inputWriterScanner(inputReader *io.PipeReader) {
+	scanner := bufio.NewScanner(inputReader)
+	for scanner.Scan() {
+		logger.Print(scanner.Text())
+	}
+	if err := scanner.Err(); err != nil {
+		logger.Fatal(err)
```

Comment:
EOF is not considered an error condition by bufio.Scanner ([doc](http://golang.org/pkg/bufio/#Scanner.Scan)). But I'll change it to non-fatal, and have it close the PipeReader so writes will fail.